chapters/extend/driver.rst

```diff
@@ -1,16 +1,181 @@
+Follow the table above to decide which `BaseDriver` class to inherit from.
+After creating your customised `Driver` class, you need to implement your own Core method based on your specific need.
+
+
+Customize Driver in Action: `MultimodalDriver`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We created `MultimodalDriver <https://github.com/jina-ai/jina/blob/master/jina/drivers/multimodal.py>`_ to better support Cross/Multi-modal search.
+Assume a `Document` has 2 child `Documents` with different modalities, e.g. text and image.
+The objective of `MultimodalDriver` is to extract embeddings from both modalities.
+Our expected input and output can be represented as:
+
+.. highlight:: shell
+.. code-block:: shell
+
+    Input:
+    document:
+            |- child document: {modality: mode1}
+            |
+            |- child document: {modality: mode2}
+    Output:
+    document: (embedding: multimodal encoding)
+            |- child document: {modality: mode1}
+            |
+            |- child document: {modality: mode2}
+
+
+In the code snippt blow, you should be able to see the logic of how we implemented the Driver.
+In :meth:`_apply_all`,
+We firstly convert each Document in the `DocumentSet` into a `MultimediaDocument`.
+For each instance of the `MultimediaDocument`,
+we check if the instance has :meth:`modality_content_map` (a python `dict`, where key is the name of the modality, while value is the content of the modality).
+If so, we consider it has a valid `MultimediaDocument`.
+And we save the value of the specific Modality into `content_by_modality`.
+The result will be feed into :meth:`exec_fn` as the input.
+
+Since class `MultiModalDriver` is based on `BaseEncodeDriver`, the :meth:`exec_fn` will be bind to the :meth:`encode` method,
+to encode data from different modalities into embeddings.
+Last but not least, we assign the `embeddings` property of each Document as the encoded vector representation.
+
+.. highlight:: python
+.. code-block:: python
+
+
+    class MultiModalDriver(FlatRecursiveMixin, BaseEncodeDriver):
+    """Extract multimodal embeddings from different modalities."""
+
+        @property
+        def positional_modality(self) -> List[str]:
+            return self._exec.positional_modality
+
+        def _get_executor_input_arguments(
+            self, content_by_modality: Dict[str, 'np.ndarray']
+        ) -> List['np.ndarray']:
+            """From a dictionary ``content_by_modality`` it returns the arguments in the proper order so that they can be
+            passed to the executor.
+             :param content_by_modality: a dictionary of `Document content` by modality name
+             :return: list of input arguments as np arrays
+            """
+            return [content_by_modality[modality] for modality in self.positional_modality]
+
+        def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
```

Comment:
I think we do not need to duplicate exactly the code, we should just add small amount of code that can explain the concept

Reply:
@JoanFM I simplified the code